features/partnerhistorie.feature

```diff
@@ -0,0 +1,110 @@
+# language: nl
+
+Functionaliteit: Tonen van Partnerhistorie
+  Huidige en beëindigde huwelijken en geregistreerd partnerschappen van ingeschreven personen kunnen worden geraadpleegd.
+
+  Alleen huwelijken en partnerschappen uit categorie 05 worden opgenomen in het antwoord.
+  Bij het ophalen van ontbonden huwelijken en partnerschappen uit GBA-V wordt Datum huwelijkssluiting/aangaan geregistreerd partnerschap (06.10) uit de eerste historische huwelijk/partnerschap (categorie 55) na het actuele huwelijk/partnerschap waarin  Datum huwelijkssluiting/aangaan geregistreerd partnerschap een waarde heeft.
```

Comment:
Ik begrijp deze zin niet, mogelijk omdat ik niet weet hoe een historische partner door GBA-V wordt geretourneerd. Klopt het dat een historische partner meerdere voorkomens kan hebben en dat deze in een bepaalde volgorde wordt geretourneerd en dat in dit geval de eerste Datum huwelijkssluiting/aangaan geregistreerd partnerschap moet worden gepakt die volgt na de actuele?

Reply:
Als ik naar de scenario's kijk, kom ik tot de conclusie dat een historische partner inderdaad met meerdere voorkomens kan worden geregistreerd.
Is het een idee om voor het mappen van GBA-V gegevens naar een BRP resource om dat in een aparte feature file te specificeren en in deze feature file alleen de specificaties voor het raadplegen? Ik denk dat de specs hierdoor smarter worden.

Reply:
Hoi Melvin, dank voor je review. Denk je dat de feature ook relevant is voor raadplegen?

Ik heb de feature geschreven vanuit het perspectief dat de provider de doelgroep is. En dat de Open API Specificaties voldoende informatie zouden moeten geven voor de API-consumer. Zie jij dit anders?

Heb je voorbeeld(en) van plekken waar de specificatie niet smart genoeg is (voor de developer van de provider)?

Reply:
Hoi Frank,

Wat ik in de feature file nu zie, is dat in de achtergrond (bijv. regel 31, 32, 33) 3 rijen zijn om een (historische) partner te definieren, maar als we het teruggeven als een resource dan is het in principe 1 rij geworden. Dit kan dan als een eigen feature (hoe mappen we historische gba gegevens naar een historische partner sub-resource) met eventueel andere scenario's worden gespecificeerd. Deze info relevant voor de provider.

In de raadpleeg partner historie feature hoeft dan alleen te worden gespecificeerd welke partner gegevens je dan in welke combinatie terug kan krijgen. Dit kan dan door zowel de consumer als providers worden gebruikt om te bepalen hoe de functionaliteit moet werken.

Het splitsen van de feature heeft verder als bijkomende voordeel dat de raadpleeg feature in combinatie met de API specificatie kan worden hergebruikt door providers die geen gba-v als bron gebruiken